flang/lib/Semantics/check-omp-structure.cpp

```diff
@@ -2531,6 +2539,9 @@ void OmpStructureChecker::CheckReductionModifier(
     switch (dirCtx.directive) {
     case llvm::omp::Directive::OMPD_do: // worksharing-loop
     case llvm::omp::Directive::OMPD_do_simd: // worksharing-loop simd
+    case llvm::omp::Directive::
```

Comment:
Should `OMPD_[[target_]teams_]distribute_parallel_do[_simd]` and `OMPD_distribute_simd` be listed here as well?

Reply:
Added